include/pybind11/cast.h

```diff
@@ -231,22 +237,19 @@ class type_caster_generic {
 
         auto &internals = get_internals();
 
-        auto it = internals.registered_types_cpp.find(std::type_index(*type_info));
-        if (it == internals.registered_types_cpp.end()) {
```

Comment:
Although less nice from a coding style view, I think that it would be nicer to leave the longer ``internals.registered_types_cpp.find`` calls here. This is a very "hot" piece of code, and the latter are more efficient since they don't have to resolve the ``internals`` data structure multiple times.